src/basic/missing_fcntl.h

```diff
@@ -43,9 +43,3 @@
 #ifndef AT_HANDLE_FID
 #define AT_HANDLE_FID AT_REMOVEDIR
 #endif
-
-/* On musl, O_ACCMODE is defined as (03|O_SEARCH), unlike glibc which defines it as
- * (O_RDONLY|O_WRONLY|O_RDWR). Additionally, O_SEARCH is simply defined as O_PATH. This changes the behaviour
- * of O_ACCMODE in certain situations, which we don't want. This definition is copied from glibc and works
- * around the problems with musl's definition. */
-#define O_ACCMODE_STRICT (O_RDONLY|O_WRONLY|O_RDWR)
```

Comment:
Sorry, but I vehemently disagree. Even glibc documents that O_PATH is an accmode, and ought to be manually excluded: https://www.gnu.org/software/libc/manual/html_node/Access-Modes.html. The argument makes zero sense to me either. You simply need to be aware of our own helpers, e.g. isatty_safe instead of isatty, close_nointr rather than close.

Reply:
Another way to fix the issue is already waiting in #37788.
As said in the commit message, the previous fix does not scale and we can easily introduce the same issue in a new code.

Regardless we should fix the issue in upstream or not, the fix needs to be reverted.

Reply:
> the previous fix does not scale and we can easily introduce the same issue in a new code

Again, that's not a valid argument. If you want to make this more observable, please update coding style to mention this macro. But we carry a bunch of downstream helpers  to patch up behaviors in kernel/libc calls, and this one is really not special if you ask me.

Reply:
Why do you refuse a better way to fix the issue? Have you seen the fix in #37788?